driver-core/src/main/com/mongodb/internal/operation/OperationHelper.java

```diff
@@ -171,6 +171,16 @@ static void validateHint(final ConnectionDescription connectionDescription, fina
         }
     }
 
+    static void validateAllowDiskUse(final AsyncConnection connection, final Boolean allowDiskUse,
```

Comment:
I think you misunderstood my previous comment about inlining the method.  I meant to inline only the method that takes a ConnectionDescription, not the one that takes a Connection.  Now it looks even worse, since there is even less consistency between the sync and async cases.  Can you take another look and try to keep them similar?